### PR TITLE
fix: add fallback and validation for report and cron timezones

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
It was possible for short period of time to save an invalid timezone value (GMT Standard Time) in the reports schedule, which was causing errors and failures when running the cron job. With this change, I am catching any cron errors with regard to the timezone function and defaulting to UTC. I also added timezone validation when creating a report. 

### TESTING INSTRUCTIONS
unit and integration tests have been updated. To manually test, you should be able to create and run a report with a timezone. You should not be able to create a report with a bad timezone. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/reports/schemas.py [hunk 1]
```diff
@@ -153,7 +154,11 @@ class ReportSchedulePostSchema(Schema):
         allow_none=False,
         required=True,
     )
-    timezone = fields.String(description=timezone_description, default="UTC")
+    timezone = fields.String(
+        description=timezone_description,
+        default="UTC",
+        validate=validate.OneOf(choices=tuple(all_timezones)),
```
Comment (betodealmeida) [inline superset/reports/schemas.py:160]: Nice!

File: superset/tasks/cron_util.py [hunk 2]
```diff
@@ -28,8 +28,12 @@ def cron_schedule_window(cron: str, timezone: str) -> Iterator[datetime]:
     window_size = app.config["ALERT_REPORTS_CRON_WINDOW_SIZE"]
     # create a time-aware datetime in utc
     time_now = datetime.now(tz=dt_timezone.utc)
-    tz = pytz.timezone(timezone)
-    utc = pytz.timezone("UTC")
+    try:
+        tz = pytz_timezone(timezone)
+    except UnknownTimeZoneError:
+        # fallback to default timezone
+        tz = pytz_timezone("UTC")
```
Comment (betodealmeida) [inline superset/tasks/cron_util.py:35]: Maybe add a log here, saying that a report was saved with an invalid timezone?

**Review by betodealmeida — APPROVED** [review]
This is great!

**Review by geido — APPROVED** [review]
LGTM!

File: superset/tasks/cron_util.py (hunk 2 above)
Comment (eschutho) [reply superset/tasks/cron_util.py:38]: good idea!

**Pull request merged**